tensorflow/python/framework/device_spec.py

```diff
@@ -356,6 +356,15 @@ def _components_to_string(job, replica, task, device_type, device_index):
     return output
 
   def __eq__(self, other):
+    """Checks if two known `DeviceSpec`s are same.
+
+    Args:
+      other: Another DeviceSpec
+
+    Returns:
+      Returns `True` if other DeviceSpec and this DeviceSpec belong to the same class and
```

Comment:
Return `True` if `other` is also a DeviceSpec instance and has same value as the current instance. Return `False` otherwise.